### DISCUSSION TITLE
Support for BCP 47

### DISCUSSION THREAD

**Discussion by brsvh** [srid/emanote#484] [Ideas]
I read the documentation of Heist template syntax and emanote templates, and I realized that emanote seems to lack some mechanisms for checking and generating multilingual content.

emanote's default template cannot rigorously generate pages for specific languages, it does not have extra handling for language tags (BCP 47). We cannot generate pages or html blocks with correct tags, for example, the base theme uses a hard-coded value.

https://github.com/srid/emanote/blob/5807f6913f0a07a36995422079c12a236177e939/emanote/default/templates/base.tpl#L2

We expect to get HTML like this in a localized website:

```html
<html lang="zh-Hans">
  <!-- ... -->
  <body>
    <p>Some text in Simplified Chinese.</p>
    <p lang="fr">Ce paragraphe est défini en français.</p>
  </body>
</html>

```

I think this might be possible by allowing explicit specification of `page.lang` to make the template get the correct value.

Perhaps consider allowing a design like this.

```
<ema:metadata>
  <with var="page">
    <html lang="${value:lang}">
  </with>
</ema:metadata>
```

This way we seem to get some extra benefits, such as using templates in different languages, like `<apply template="components/foo/bar.${value:lang}.tpl" />`.

**Comment by srid** [comment]
I hope this will do it: https://github.com/srid/emanote/pull/485
